peachjam/models/core_document_model.py

```diff
@@ -911,16 +911,18 @@ def update_or_create_for_document(cls, document):
             text = " ".join(root.itertext())
 
         elif hasattr(document, "source_file"):
-            # get the text from the source file, via PDF if necessary
-            with tempfile.NamedTemporaryFile() as tmp:
-                # convert document to pdf and then extract the text
-                pdf = document.source_file.as_pdf()
-                shutil.copyfileobj(pdf, tmp)
-                tmp.flush()
-                text = pdfjs_to_text(tmp.name)
-                # some PDFs have nulls, which breaks SQL insertion
-                # replace rather than deleting to keep string length the same
-                text = text.replace("\0", " ")
+            if document.source_file.pk:
+                # get the text from the source file, via PDF if necessary
```

Comment:
during an update the _savem2m sometimes get called before saving the source file, i think when saving other models?